### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by mwellick**

### CODE REVIEW

File: app/main.py [hunk 1]
```diff
@@ -1,3 +1,47 @@
...
+    def __add__(self, other: Union["Distance", int | float]) -> "Distance":
+        if isinstance(other, Distance):
+            return Distance(self.km + other.km)
+        elif isinstance(other, (int | float)):
+            return Distance(self.km + other)
+
+    def __iadd__(self, other: Union["Distance", int | float]) -> "Distance":
+        if isinstance(other, Distance):
+            self.km += other.km
+        elif isinstance(other, int | float):
```
Comment (Bohdan-Salamakha) [inline app/main.py:24]: remove redundant "elif" check

File: app/main.py [hunk 2]
```diff
@@ -1,3 +1,47 @@
...
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other: Union["Distance", int | float]) -> "Distance":
+        if isinstance(other, Distance):
+            return Distance(self.km + other.km)
+        elif isinstance(other, (int | float)):
```
Comment (Bohdan-Salamakha) [inline app/main.py:18]: remove redundant "elif" check

**Review by EdAlekseiev — CHANGES_REQUESTED** [review]

File: app/main.py [hunk 3]
```diff
@@ -1,3 +1,58 @@
...
+            self.km += other.km
+        else:
+            self.km += other
+
+        return self
+
+    def __mul__(self, other: Union["Distance", int | float]) -> "Distance":
+        return Distance(self.km * other)
+
+    def __truediv__(self, other: Union["Distance", int | float]) -> "Distance":
```
Comment (olena-hul) [inline app/main.py:32]: other can be only int or float in __truediv__

File: app/main.py [hunk 4]
```diff
@@ -1,3 +1,58 @@
...
+
+    def __iadd__(self, other: Union["Distance", int | float]) -> "Distance":
+        if isinstance(other, Distance):
+            self.km += other.km
+        else:
+            self.km += other
+
+        return self
+
+    def __mul__(self, other: Union["Distance", int | float]) -> "Distance":
```
Comment (olena-hul) [inline app/main.py:29]: other can be only int or float in __mul__

**Review by olena-hul — CHANGES_REQUESTED** [review]

**Review by Bohdan-Salamakha — APPROVED** [review]

**Pull request closed without merging**